lib/polymorphic_integer_type/extensions.rb

```diff
@@ -24,26 +24,32 @@ def belongs_to(name, scope = nil, **options)
             mapping
           end
 
-          define_method foreign_type do
-            t = super()
-            mapping[t]
-          end
+          component_extension = Module.new do
+            define_method foreign_type do
+              t = super()
+              self.class.send("#{foreign_type}_mapping")[t]
+            end
 
-          define_method "#{foreign_type}=" do |klass|
-            enum = mapping.key(klass.to_s)
-            if klass.kind_of?(Class) && klass <= ActiveRecord::Base
-              enum ||= mapping.key(klass.base_class.to_s)
-              enum ||= mapping.key(klass.base_class.sti_name)
+            define_method "#{foreign_type}=" do |klass|
+              mapping = self.class.send("#{foreign_type}_mapping")
+              enum = mapping.key(klass.to_s)
+              if klass.kind_of?(Class) && klass <= ActiveRecord::Base
+                enum ||= mapping.key(klass.sti_name)
```

Comment:
This line is also new. 

Previously, we checked `klass.to_s`, then `klass.base_class.to_s`, then `klass.base_class.sti_name`. This ensures that we check `klass.sti_name` before we move to the base class.